readthedocs/restapi/utils.py

```diff
@@ -18,17 +20,19 @@
 
 def sync_versions(project, versions, type):  # pylint: disable=redefined-builtin
     """Update the database with the current versions from the repository."""
-    old_versions = {}
-    old_version_values = project.versions.filter(type=type).values(
-        'identifier', 'verbose_name')
-    for version in old_version_values:
-        old_versions[version['verbose_name']] = version['identifier']
+    old_version_values = project.versions.filter(type=type).values_list(
+        'verbose_name', 'identifier'
+    )
+    old_versions = dict(old_version_values)
 
-    added = set()
     # Add new versions
+    added = set()
+    has_user_stable = False
     for version in versions:
         version_id = version['identifier']
         version_name = version['verbose_name']
+        if version_name == STABLE_VERBOSE_NAME:
+            has_user_stable = True
```

Comment:
Since we are using `has_user_stable` outside this loop, I think it's preferable to create it right before where it's used: line 63 using.

Reply:
But is still outside the loop, so it's really not right before it's used.

Reply:
Yes. Is is used inside the loop?

Reply:
Nop, but the logic that came behind is not related to that code.